### DISCUSSION TITLE
erro ao obter via horse

### DISCUSSION THREAD

**Discussion by FranziniDev** [HashLoad/horse#372] [General]
Ao tentar instalar via horse, obtenho o erro abaixo: 

C:\fontes\api\ControleServidor>boss install horse
[INFO ] Installing modules in project path
[INFO ] Building cache files...
[INFO ] Updating cache of dependency horse
[INFO ] Installing modules
[INFO ] Processing dependency horse
[WARN ] Tag not found , using semantic now...
[WARN ]   No candidate to version for >0.0.0. Using master branch
[ERROR]   Error on switch to needed version from dependency github.com/hashload/horse
reference not found

**Comment by viniciussanchez** [comment]
Como esta o seu arquivo boss.json / boss-lock.json?
O boss está atualizado?

**Reply by FranziniDev** [reply]
C:\fontes\api\ControleServidor>boss -v
v3.0.10

boss.josn :

{
"name": "ControleServidor",
"description": "",
"version": "1.0.0",
"homepage": "",
"mainsrc": "./src",
"projects": [],
"dependencies": {}
}

boss-lock :

{
"hash": "d41d8cd98f00b204e9800998ecf8427e",
"updated": "2024-01-11T08:42:59.6370208-03:00",
"installedModules": {}
}

Já apaguei eles e comecei de novo, mas assim mesmo tenho aquela resposta. Será algo com o Git ?

**Reply by viniciussanchez** [reply]
Estranho... acabei de fazer o teste e deu certo aqui:

![image](https://github.com/HashLoad/horse/assets/16382981/98a8ccda-32ee-4700-ac01-ec360eac4172)

**Reply by FranziniDev** [reply]
Vinicius, vou rever aqui meu firewall, o que achei estranho, é que em projetos que já tinham o horse instalado via boss, tambem estou tendo erro: 

C:\fontes\api\Concentrador>boss update
[INFO ] Installing modules in project path
[INFO ] Building cache files...
[INFO ] Updating cache of dependency horse
[INFO ] Updating cache of dependency horse-basic-auth
[INFO ] Downloading dependency github.com/hashload/horse-basic-auth
[INFO ] Updating cache of dependency horse-cors
[INFO ] Downloading dependency github.com/hashload/horse-cors
[INFO ] Installing modules
[INFO ] Processing dependency horse
[WARN ]   No candidate to version for ^3.1.5. Using master branch
[ERROR]   Error on switch to needed version from dependency github.com/hashload/horse
reference not found

Existe a possibilidade de que de alguma forma o Git esteja me bloqueando ?  O Boos ja esta na lista de permissões do meu firewall.

**Reply by FranziniDev** [reply]
Vinicius, me desculpe por tomar seu tempo, obrigado pela resposta rapida. 
Consegui aqui, não faz sentido nenhum, mas vou relatar aqui, o curioso acontecido. 
Estava desde mais cedo tentando sem sucesso. Agora neste momento mudei a ordem de instalação, executei primeiro o comendo : 

C:\fontes\api\Concentrador>boss install github.com/viniciussanchez/dataset-serialize 

Neste obtive sucesso de primeira, logo em seguinda confirmei o comando : 

C:\fontes\api\Concentrador>boss install horse

A partir dai esta respondendo, instalando e atualizando.

**Reply by viniciussanchez** [reply]
Estranho kkkkkkk
mas se resolveu blz
